test/e2e/config_test.go

```diff
@@ -0,0 +1,65 @@
+package e2e
+
+import (
+	"context"
+	"fmt"
+	"testing"
+
+	"github.com/stretchr/testify/require"
+	corev1 "k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+
+	configapi "github.com/k8ssandra/k8ssandra-operator/apis/config/v1beta1"
+	"github.com/k8ssandra/k8ssandra-operator/test/framework"
+)
+
+func controllerRestart(t *testing.T, ctx context.Context, namespace string, f *framework.E2eFramework) {
```

Comment:
It would be nice to have this (or a different) test deploy a K8ssandraCluster that uses the new, valid ClientConfig. That will provide good coverage over `main.go`.

Reply:
I think main.go is tested every time any test runs? Wouldn't we fail to connect to other clusters otherwise?

Reply:
Good point